### PR TITLE
bug fix: shutdown storage server may core, if internal storage thread is not stopped. & some toss bug fix

### PR DESCRIPTION
**Pull request by liuyu85cn**
1. shutdown storage server may core, if internal storage thread is not stopped
2. toss bug fix:
    2.1 toss iter will not skip invalid resume lock
    2.2 toss iter won't wait all resume task finished
    2.3 revese key get wrong part in txn manager 

depends on https://github.com/vesoft-inc/nebula-common/pull/352

### CODE REVIEW

File: src/storage/transaction/TossEdgeIterator.h [hunk 1]
```diff
@@ -167,24 +172,34 @@ class TossEdgeIterator : public SingleEdgeIterator {
...
+            LOG_IF(INFO, FLAGS_trace_toss) << "next(), waiting resume finished";
+            folly::collectAll(std::move(resumeTasks_)).wait();
             needWaitResumeTask_ = false;
             recoverEdgesIter_ = recoverEdges_.begin();
         } else {
             recoverEdgesIter_++;
         }
 
-        if (recoverEdgesIter_ != recoverEdges_.end()) {
+        while (recoverEdgesIter_ != recoverEdges_.end()) {
```
Comment (critical27) [inline src/storage/transaction/TossEdgeIterator.h:188]: Good catch

File: src/storage/transaction/TransactionManager.cpp [hunk 2]
```diff
@@ -185,14 +185,18 @@ folly::Future<cpp2::ErrorCode> TransactionManager::addSamePartEdges(
                     // steps 4 & 5: multi put local edges & multi remove persist locks
                     kvstore::BatchHolder bat;
                     for (auto& lock : lockData) {
-                        bat.remove(NebulaKeyUtils::toLockKey(lock.first));
+                        bat.remove(std::string(lock.first));
```
Comment (critical27) [inline src/storage/transaction/TransactionManager.cpp:188]: Maybe could move

File: src/storage/transaction/TransactionManager.cpp [hunk 3]
```diff
@@ -391,7 +390,7 @@ folly::SemiFuture<kvstore::ResultCode> TransactionManager::commitEdgeOut(size_t
     if (!indexes.empty()) {
         std::vector<kvstore::KV> data;
         data.reserve(32);
-        data.emplace_back(std::move(key), std::move(props));
+        data.emplace_back(std::string(key), std::string(props));
```
Comment (critical27) [inline src/storage/transaction/TransactionManager.cpp:393]: Why we can't move here?

**Review by critical27 — COMMENTED** [review]
Generally LGTM

File: src/storage/transaction/TransactionManager.cpp (hunk 2 above)
Comment (liuyu85cn) [reply src/storage/transaction/TransactionManager.cpp]: fixed

File: src/storage/transaction/TransactionManager.cpp (hunk 3 above)
Comment (liuyu85cn) [reply src/storage/transaction/TransactionManager.cpp]: fixed

**Review by critical27 — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]
excellent!

**Pull request merged**